src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -191,11 +182,16 @@ private bool VisitEnumerableType()
         {
             var isValid = true;
 
-            if (_model != null)
+            if (_model != null && _metadata.ValidateChildren)
             {
                 var strategy = _strategy ?? DefaultCollectionValidationStrategy.Instance;
                 isValid = VisitChildren(strategy);
             }
+            else if (_model != null)
+            {
+                // Suppress validation for the prefix, but we still want to validate the object.
```

Comment:
Won't `SuppressValidation()` also set the model's entry to `Skipped`? Put another way, is the comment intended to mean the exact match for `_key` will be updated a few lines later?